exercises/stack_01.js

```diff
@@ -0,0 +1,82 @@
+window.Collections = {};
+
+(function(module) {
+    function Stack(arr) {
+        /* constructor */
+        this.arr = arr;
+    };
+
+    /* creating a forEach method on Stack */
+    Stack.prototype.forEach = function(callback) {
+        for (var i = 0; i < this.arr.length; i++) {
+            callback(this.arr[i], i, this.arr);
```

Comment:
Remember our discussion: this needs to be `callback.call(this, ...)` to preserve `this` appropriately.